web/src/narrow_state.ts

```diff
@@ -8,15 +8,9 @@ import * as stream_data from "./stream_data";
 import type {StreamSubscription} from "./sub_store";
 import * as unread from "./unread";
 
-export let has_shown_message_list_view = false;
-
 export function filter(): Filter | undefined {
-    // `All messages`, `Recent Conversations` and `Inbox` return undefined;
-    if (message_lists.current === undefined || message_lists.current.data.filter.is_in_home()) {
-        return undefined;
-    }
-
-    return message_lists.current.data.filter;
+    // `Recent Conversations` and `Inbox` return undefined;
+    return message_lists.current?.data.filter;
 }
```

Comment:
Note that we no longer return `undefined` for all messages view since it is like any other narrow for us now.